lib/integrations/awsoidc/listdatabases_iam_config.go

```diff
@@ -0,0 +1,125 @@
+	// Region is the AWS Region.
+	// Used to set up the AWS SDK Client.
+	Region string
+
+	// IntegrationRole is the Integration's AWS Role used by the integration.
+	IntegrationRole string
+
+	// ListDatabasesPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to ListDatabases
+	ListDatabasesPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *ListDatabasesIAMConfigureRequest) CheckAndSetDefaults() error {
+	if r.Region == "" {
+		return trace.BadParameter("region is required")
+	}
+
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.ListDatabasesPolicy == "" {
+		r.ListDatabasesPolicy = defaultPolicyNameForListDatabases
+	}
+
+	return nil
+}
+
+// ListDatabasesIAMConfigureClient describes the required methods to create the IAM Policies required for Listing Databases.
+type ListDatabasesIAMConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+type defaultListDatabasesIAMConfigureClient struct {
+	*iam.Client
+}
+
+// NewListDatabasesIAMConfigureClient creates a new ListDatabasesIAMConfigureClient.
+func NewListDatabasesIAMConfigureClient(ctx context.Context, region string) (ListDatabasesIAMConfigureClient, error) {
+	if region == "" {
+		return nil, trace.BadParameter("region is required")
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &defaultListDatabasesIAMConfigureClient{
+		Client: iam.NewFromConfig(cfg),
+	}, nil
+}
+
+// ConfigureListDatabasesIAM set ups the policy required for accessing an RDS DB Instances and RDS DB Clusters.
+// It creates an inline policy with the following permissions:
+//   - rds:DescribeDBInstances
+//   - rds:DescribeDBClusters
+//
+// The following actions must be allowed by the IAM Role assigned in the Client.
+//   - iam:PutRolePolicy
+func ConfigureListDatabasesIAM(ctx context.Context, clt ListDatabasesIAMConfigureClient, req ListDatabasesIAMConfigureRequest) error {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	listDatabasesPolicyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementForListRDSDatabases(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.ListDatabasesPolicy,
+		RoleName:       &req.IntegrationRole,
+		PolicyDocument: &listDatabasesPolicyDocument,
+	})
+	if err != nil {
+		if trace.IsNotFound(awslib.ConvertIAMv2Error(err)) {
+			return trace.NotFound("role %q not found.", req.IntegrationRole)
+		}
+		return trace.Wrap(err)
+	}
+
+	log.Printf("IntegrationRole: IAM Policy %q added to Role %q\n", req.ListDatabasesPolicy, req.IntegrationRole)
```

Comment:
This is used within the context of a `teleport ....` command with a really short life span and only a couple of log lines.
For those cases, I tend to use as less external dependencies as possible.

Is there a guidance on which one to use?
If so, can we add a `depguard` rule to prevent its usage?

Reply:
Moving to logrus :+1:

Reply:
New log message: 
```
> teleport integration configure listdatabases-iam --aws-region=eu-west-2 --role=MarcoTestAWSOIDCSetup
INFO             Added Inline Policy to IAM Role policy:ListDatabases role:MarcoTestAWSOIDCSetup awsoidc/listdatabases_iam_config.go:97
```